### PR TITLE
Update HaveIBeenPwned to include API Key

### PR DESCRIPTION
**Pull request by aetaric**
**Description:**
Updates Documentation for HaveIBeenPwned to include the newly required API key per: https://www.troyhunt.com/authentication-and-the-have-i-been-pwned-api/

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#25495

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/9977"></a>

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_components/haveibeenpwned.markdown [hunk 1]
```diff
@@ -24,13 +30,18 @@ sensor:
     email:
       - your_email1@domain.com
       - your_email2@domain.com
+    api_key: apikeygoeshere
```
Comment (klaasnicolaas) [inline source/_components/haveibeenpwned.markdown]:
```suggestion
    api_key: API_KEY
```

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_components/haveibeenpwned.markdown (hunk 1 above)
Comment (frenck) [reply source/_components/haveibeenpwned.markdown]: Addressed

**Review by frenck — APPROVED** [review]

**Comment by aetaric** [comment]
The parent PR is now live in 0.97.1

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Manually cherry-picked to our current documentation.

**Pull request merged**